src/parser/shaman/restoration/modules/spells/ChainHeal.js

```diff
@@ -4,24 +4,76 @@ import SpellIcon from 'common/SpellIcon';
 import SpellLink from 'common/SpellLink';
 import SPELLS from 'common/SPELLS';
 import { TooltipElement } from 'common/Tooltip';
+import SPECS from 'game/SPECS';
+import SpecIcon from 'common/SpecIcon';
+import { formatNth, formatDuration } from 'common/format';
 
-import Analyzer from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
 
 import AbilityTracker from 'parser/shared/modules/AbilityTracker';
-
+import Combatants from 'parser/shared/modules/Combatants';
 import StatisticBox, { STATISTIC_ORDER } from 'interface/others/StatisticBox';
 
 const CHAIN_HEAL_TARGET_EFFICIENCY = 0.97;
+const HEAL_WINDOW_MS = 250;
 
 class ChainHeal extends Analyzer {
   static dependencies = {
     abilityTracker: AbilityTracker,
+    combatants: Combatants,
   };
 
+  buffer = [];
+  chainHeals = [];
+  cast = 0;
+  chainHealTimestamp = 0;
+
   constructor(...args) {
     super(...args);
     this.maxTargets = 4;
     this.suggestedTargets = this.maxTargets * CHAIN_HEAL_TARGET_EFFICIENCY;
+
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.CHAIN_HEAL), this.chainHeal);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.CHAIN_HEAL), this.chainHeal);
+  }
+
+  chainHeal(event) {
+    if (!this.chainHealTimestamp || event.timestamp - this.chainHealTimestamp > HEAL_WINDOW_MS) {
+      this.processBuffer();
+      this.chainHealTimestamp = event.timestamp;
+    }
+
+    this.buffer.push({
+      ...event,
+    });
+  }
+
+  processBuffer() {
+    if (this.buffer.length === 0) {
+      return;
+    }
+    this.cast++;
+    this.chainHeals[this.cast] = {};
+    const cast = this.buffer.find(event => event.type === "cast");
```

Comment:
Looks really great, but variable names are a bit confusing. Using `cast`, `this.cast` and `this.casts` is probably not a good idea.